adapter/src/dbt/adapters/fal_experimental/adapter_support.py

```diff
@@ -130,6 +155,14 @@ def read_relation_as_df(adapter: BaseAdapter, relation: BaseRelation) -> pd.Data
     else:
         with new_connection(adapter, "fal:read_relation_as_df") as connection:
             alchemy_engine = _get_alchemy_engine(adapter, connection)
+            if adapter.type() == "athena":
+                # This is dbt-athena-community quirk, table_name=relation.schema
+                return pd.read_sql_table(
+                    con=alchemy_engine,
+                    table_name=relation.schema,
+                    schema=relation.database,
+                )
+
```

Comment:
All of this change should live in dbt.adapters.fal_experimental.support.athena module like we do for duckdb, biugquery, etc